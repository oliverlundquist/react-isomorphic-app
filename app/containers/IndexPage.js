import React from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { FormattedNumber, FormattedMessage } from 'react-intl';

class IndexPage extends React.Component {
	getStyles(key) {
		return {
			loginBtn: {
				width: 360,
				height: 'auto'
			}
		}[key];
	}

	render() {
		return (
			<div>
				<FormattedMessage id="IndexPage.greeting" />
				<FormattedNumber value="100000000" />
				<h1>Index Page</h1>
				<a href={this.props.authUrl}><img style={this.getStyles('loginBtn')} src="https://s3-eu-west-1.amazonaws.com/mystore-api-auth-app/assets/mystore-api-login-btn.jpg" alt="Login with Mystore.no" /></a>
			</div>
		);
	}
}

IndexPage.defaultProps = {
	authUrl: 'http://passport.dev:8000/oauth/authorize?client_id=7&redirect_uri=http://consumer.dev:8010/callback&response_type=code&scope=read:products read:categories'
};

// IndexPage.propTypes = {
  // name: React.PropTypes.string
// };

const mapStateToProps = (state) => ({
	//
})

function mapDispatchToProps(dispatch) {
	return bindActionCreators({}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(IndexPage)
