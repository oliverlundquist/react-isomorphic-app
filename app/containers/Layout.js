import React from 'react';
import { Link } from 'react-router';

export class Layout extends React.Component {
	render() {
		return (
			<div className="app-container">
				<header>
					<Link to="/">Home</Link>
					<span style={{margin:'0 5px'}}>|</span>
					<Link to="/users">Users</Link>
				</header>
				<div className="app-content">{this.props.children}</div>
				<footer style={{borderTop:'1px solid #000',marginTop:'15px'}}>
					<p>This is where the footer goes.</p>
				</footer>
			</div>
		);
	}
}
