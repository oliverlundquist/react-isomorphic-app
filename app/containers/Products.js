import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { readEndpoint, setHeaders, setEndpointHost, setEndpointPath, setAccessToken } from 'redux-json-api-without-credentials'

class Products extends React.Component {
	componentWillMount() {
		// this.props.dispatch(setAccessToken(this.props.tokens.access_token))
		// this.props.dispatch(setHeaders({
			// 'Authorization': 'Bearer ' + this.props.tokens.access_token,
			// 'Authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6ImUxNWQ5MTJhNmMzY2IyZDcwOWFlZDM5ZjUwNTE0YWFlNjhhNTM5ODhhNGVmMTEwOWJlNDI0ZWIwYWIwMzA0M2JhM2M4NWU3NTM4MGFhYTcxIn0.eyJhdWQiOiIyIiwianRpIjoiZTE1ZDkxMmE2YzNjYjJkNzA5YWVkMzlmNTA1MTRhYWU2OGE1Mzk4OGE0ZWYxMTA5YmU0MjRlYjBhYjAzMDQzYmEzYzg1ZTc1MzgwYWFhNzEiLCJpYXQiOjE0NzU3NTE1ODcsIm5iZiI6MTQ3NTc1MTU4NywiZXhwIjo0NjMxNDI4Nzg3LCJzdWIiOiIyIiwic2NvcGVzIjpbIioiXSwibm9fbXlzdG9yZV9ob3N0cyI6WyJjaGlwcyJdfQ.ep7f7aJgfdwnmBQMXrTHf-aMMYjx3pS4wCKOJ1tp42A9HehJTXXWq7vExxoziy5FmMvqOlvh9UdxUkpooz76M8VPxaFzkp3RRXM938AP_zKgGAv9wliQLAIGNE1tx1hlbaRzf6q6lzbWKbi-rb45GOc8C7CrR4rzpuJu8HbJk0ezovpy210lTuEl5aIWMHGqpNSLyjuN6tYHF3vAnKaj8Z9936sWy4KfHfTuWAqBDJwtTyt8WuwtlxA0fRGBePe8bSy7KaTskP2FmysAizbn0Tga0bADmvtlrarcrzMu--5ZxEGUad8D5nCI-FVNiLXPvei5X8TSxXl1HfRVob8PVlcbmB7iA5ogNfl3czGh_EFoO3w_0AFQ1sX6aYxqfoKKoqIsd0kR04kMFA9yE0TlOyDlUt6xpRmjAnXFuX2Hk5SUZ10HdfEC5Yo8tFobzOi1PjQktGlEPrQBMojUJk05otYFJ8YqSjt9Z0xOQhnnPyN4ZiD4XhV5RVBt7kTIX2QSG6Z0BzDcgOsSS0gCrcb3SMciE5VvHSc7N2TLVXM20fCyYRrtXLxpejuois_bKkKOLQ30Ch7eC-_t0QdSEC-XfIzVTbYdx_FWymwRmz_ASthuy2Zr_KQQSNI4XY7gu4gtMUQBYCG48D5Zp8Qh-c37r56EM_Sa9fnqLVZmfWrsJlo',
			// 'Accept': 'application/vnd.api+json',
			// 'Content-Type': 'application/vnd.api+json',
			// 'Access-Control-Allow-Headers': 'Authorization'
		// }))
		// this.props.dispatch(setHeader({ 'host': '0.0.0.0' }))
		// this.props.dispatch(setEndpointHost('https://api.mystore.no'))
		// this.props.dispatch(setEndpointPath(''))
		// this.props.dispatch(readEndpoint('/shops/chips/products'))

		// <li key={product.id}><img src={'http://cdn.mystore4.no/cdn/70_57/chips/' + product.attributes.image} alt="" /></li>
	}

	render() {
		return (
			<div>
				Products!
				<ul>
					{this.props.products.data.map(product => (
						<li key={product.id}>{product.attributes.price}</li>
					))}
				</ul>
			</div>
		);
	}
}

// Users.propTypes = {
  // name: React.PropTypes.string
// };

// const mapStateToProps = ({ api: { products = { data: [] } } }) => ({ products });

// const mapStateToProps = (state) => ({
// 	api: state.api
// })

const mapStateToProps = ({
		api: { products = { data: [] } },
		session: { tokens }
	}) => ({ products, tokens })

// const mapStateToProps = (state) => ({
// 	tokens: state.session.tokens,
// 	api: state.api
// }) => ({ tokens, api })

function mapDispatchToProps(dispatch) {
	return bindActionCreators({}, dispatch)
}

// export default connect(mapStateToProps, mapDispatchToProps)(Products)
export default connect(mapStateToProps)(Products)
