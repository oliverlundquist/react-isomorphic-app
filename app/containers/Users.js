import React from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as Components from '../components'
import * as Actions from '../actions'

class Users extends React.Component {
	render() {
		return (
			<div>
				<h1>Users Page</h1>
				<div>State inside the Users Container: {this.props.count}</div>
				<hr />
				<div>Counter Component:</div>
				<Components.Counter />
			</div>
		);
	}
}

// Users.propTypes = {
  // name: React.PropTypes.string
// };

const mapStateToProps = (state) => ({
	count: state.counter.count
})

function mapDispatchToProps(dispatch) {
	return bindActionCreators({}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Users)
