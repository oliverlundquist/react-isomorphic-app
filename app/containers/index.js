export { Layout } from './Layout'
export { NotFoundPage } from './NotFoundPage'
export { default as IndexPage } from './IndexPage'
export { default as Users } from './Users'
