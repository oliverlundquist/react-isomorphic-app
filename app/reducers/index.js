export { default as counter } from './CountReducer'
export { default as session } from './SessionReducer'
