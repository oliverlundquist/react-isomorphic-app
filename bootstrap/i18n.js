import { addLocaleData } from 'react-intl'
import en from 'react-intl/locale-data/en'
import ja from 'react-intl/locale-data/ja'
import * as messages from '../lang'

addLocaleData([...en, ...ja]);

export const locales = ['en', 'ja']
export const localeIsValid = locale => locales.indexOf(locale) !== -1
export const getMessages = locale => messages[locale] || {}
