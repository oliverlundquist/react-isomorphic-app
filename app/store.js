import React from 'react';
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { routerReducer } from 'react-router-redux'
import createLogger from 'redux-logger'
import invariant from 'redux-immutable-state-invariant'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'remote-redux-devtools'
import * as reducers from './reducers'
import { reducer as api } from 'redux-json-api-without-credentials'

export const configureStore = (initialState) => {
	// TODO: move logic to external module and refine validation logic
	const environment = process.env.NODE_ENV || 'development'
	const isClient = () => typeof window !== 'undefined' && environment !== 'testing'
	const isServer = () => ! isClient()

	const enhancer = (environment === 'development' && isClient() === true)
		? composeWithDevTools({ realtime: true, port: 3001 })(applyMiddleware(invariant(), createLogger(), thunk))
		: applyMiddleware(thunk)

	return createStore(
		combineReducers({ routing: routerReducer, api: api, ...reducers }),
		initialState,
		enhancer
	)
}
