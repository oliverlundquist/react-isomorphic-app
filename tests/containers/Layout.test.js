import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { Layout, IndexPage } from '../../app/containers';
import { configureStore } from '../../app/store'
import { IntlProvider } from 'react-intl'
import { locales, getMessages } from '../../bootstrap/i18n'

test('Layout with childview', () => {
	const store = configureStore({ counter: { count: 5 } })
	const component = renderer.create(
		<Provider store={store}>
			<IntlProvider locale={locales[0]} messages={getMessages(locales[0])}>
				<Layout>
					<IndexPage />
				</Layout>
			</IntlProvider>
		</Provider>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Layout without childview', () => {
	const store = configureStore({ counter: { count: 5 } })
	const component = renderer.create(
		<Provider store={store}>
			<IntlProvider locale={locales[0]} messages={getMessages(locales[0])}>
				<Layout />
			</IntlProvider>
		</Provider>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
