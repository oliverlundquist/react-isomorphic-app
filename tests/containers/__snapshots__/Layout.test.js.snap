exports[`test Layout with childview 1`] = `
<div
  className="app-container">
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "body{margin: 0;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;}html{background: #eee;
      font-size: 100%;}@media (min-width: 550px){html{font-size: 120%;}}@media (min-width: 1200px){html{font-size: 140%;}}h1, h2, h3{font-weight: bold;}",
      }
    } />
  <header>
    <a
      onClick={[Function]}
      style={Object {}}>
      Home
    </a>
    <span
      style={
        Object {
          "margin": "0 5px",
        }
      }>
      |
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      Users
    </a>
    <span
      style={
        Object {
          "margin": "0 5px",
        }
      }>
      |
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      Products
    </a>
    <a
      href="/products">
      Products (direct linkz)
    </a>
  </header>
  <div
    className="app-content">
    <div>
      <span>
        hello
      </span>
      <h1>
        Index Page
      </h1>
      <a
        href="http://passport.dev:8000/oauth/authorize?client_id=7&redirect_uri=http://consumer.dev:8010/callback&response_type=code&scope=read:products read:categories">
        <img
          alt="Login with Mystore.no"
          src="https://s3-eu-west-1.amazonaws.com/mystore-api-auth-app/assets/mystore-api-login-btn.jpg"
          style={
            Object {
              "height": "auto",
              "width": 360,
            }
          } />
      </a>
    </div>
  </div>
  <footer
    style={
      Object {
        "borderTop": "1px solid #000",
        "marginTop": "15px",
      }
    }>
    <p>
      This is where the footer goes.
    </p>
  </footer>
</div>
`;

exports[`test Layout without childview 1`] = `
<div
  className="app-container">
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "body{margin: 0;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;}html{background: #eee;
      font-size: 100%;}@media (min-width: 550px){html{font-size: 120%;}}@media (min-width: 1200px){html{font-size: 140%;}}h1, h2, h3{font-weight: bold;}",
      }
    } />
  <header>
    <a
      onClick={[Function]}
      style={Object {}}>
      Home
    </a>
    <span
      style={
        Object {
          "margin": "0 5px",
        }
      }>
      |
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      Users
    </a>
    <span
      style={
        Object {
          "margin": "0 5px",
        }
      }>
      |
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      Products
    </a>
    <a
      href="/products">
      Products (direct linkz)
    </a>
  </header>
  <div
    className="app-content" />
  <footer
    style={
      Object {
        "borderTop": "1px solid #000",
        "marginTop": "15px",
      }
    }>
    <p>
      This is where the footer goes.
    </p>
  </footer>
</div>
`;
